test/e2e/proxy_publicviewer_test.go

```diff
@@ -0,0 +1,289 @@
+		sb := CreateCommunitySpaceBinding(t, hostAwait, space.Name, space.Namespace)
+		require.NotNil(t, sb)
+
+		// Wait until space is flagged as community
+		_, err := hostAwait.WaitForSpaceBinding(t, toolchainv1alpha1.KubesawAuthenticatedUsername, space.Name, wait.UntilSpaceBindingHasSpaceRole("viewer"))
+		require.NoError(t, err)
+
+		// wait until the space has ProvisionedNamespaces
+		sp, err := hostAwait.WaitForSpace(t, space.Name, wait.UntilSpaceHasAnyProvisionedNamespaces())
+		require.NoError(t, err)
+
+		t.Run("user is not banned", func(t *testing.T) {
+			for s, c := range tt {
+				t.Run(s, func(t *testing.T) {
+					// build proxy client
+					user := c.proxyClientUser()
+					proxyWorkspaceURL := hostAwait.ProxyURLWithWorkspaceContext(sp.Name)
+					communityUserProxyClient, err := hostAwait.CreateAPIProxyClient(t, user.Token(), proxyWorkspaceURL)
+					require.NoError(t, err)
+
+					t.Run("can list config maps", func(t *testing.T) {
+						cms := corev1.ConfigMapList{}
+						err = communityUserProxyClient.List(context.TODO(), &cms, client.InNamespace(sp.Status.ProvisionedNamespaces[0].Name))
+						require.NoError(t, err)
+					})
+
+					t.Run("cannot create config maps", func(t *testing.T) {
+						cm := corev1.ConfigMap{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "test-cm",
+								Namespace: sp.Status.ProvisionedNamespaces[0].Name,
+							},
+						}
+						err = communityUserProxyClient.Create(context.TODO(), &cm)
+						require.True(t, errors.IsForbidden(err), "expected Create ConfigMap as community user to return a Forbidden error, actual: %v", err)
+					})
+				})
+			}
+		})
+
+		t.Run("user is banned", func(t *testing.T) {
+			for s, c := range tt {
+				t.Run(s, func(t *testing.T) {
+					user := c.proxyClientUser()
+					banUser(t, hostAwait, user)
+
+					t.Run(s, func(t *testing.T) {
+						url := hostAwait.ProxyURLWithWorkspaceContext(sp.Name)
+						proxyClient, err := hostAwait.CreateAPIProxyClient(t, user.Token(), url)
+						require.NoError(t, err)
+
+						// TODO(@filariow): enable this once registration-service's middleware is implemented
+						//
+						// t.Run("cannot list config maps", func(t *testing.T) {
+						// 	cms := corev1.ConfigMapList{}
+						// 	err := proxyClient.List(context.TODO(), &cms, client.InNamespace(sp.Status.ProvisionedNamespaces[0].Name))
+						// 	require.Zero(t, cms)
+						// 	require.Error(t, err)
+						// })
+						t.Run("cannot create config maps", func(t *testing.T) {
+							cm := corev1.ConfigMap{
+								ObjectMeta: metav1.ObjectMeta{
+									Name:      "test-cm",
+									Namespace: sp.Status.ProvisionedNamespaces[0].Name,
+								},
+							}
+							err := proxyClient.Create(context.TODO(), &cm)
+							require.True(t, errors.IsForbidden(err), "expected Create ConfigMap as SSO user to return a Forbidden error, actual: %v", err)
+						})
+					})
+				})
+			}
+		})
+	})
+
+	t.Run("space is not flagged as community", func(t *testing.T) {
```

Comment:
isn't this already covered in the existing https://github.com/codeready-toolchain/toolchain-e2e/blob/58eee93627f35d2d9b67def93d9c72232facbb87/test/e2e/parallel/proxy_test.go ?

Reply:
Those tests are testing user access to workspaces when Public-Viewer is disabled. This one is testing user access to private workspaces with Public-Viewer enabled

Reply:
Ok so this makes sure that the feature toggle doesn't affect the current functionality for private workspaces. Thanks for explaining that, I've missed it.

Reply:
I would maybe just add a check to make sure the `space` doesn't have the `KubesawAuthenticatedUsername` spacebinding before starting with all the other subtests, unless I've missed it , this doesn't seem to be there now.

Reply:
good idea, it's not there. Let me add it.

Reply:
eea5b2d5df84b7ec82338430ba94e4b8e971bd7e